Fixes for HoloLens builds including Vuforia and OpenXR update closing #1170

robhillman97 opened the pull request:
Includes;
- Update to use OpenXR
- Update to Vuforia 10.15.4
- Addition of world space console window
- Multiple fixes for various HoloLens issues detailed [here](https://github.com/WEKIT-ECS/MIRAGE-XR/milestone/17)

robhillman97 committed:
fixed error in KeywordService.cs

robhillman97 committed:
fix for mask rect issue on Hololens

robhillman97 committed:
OpenXR plugin added

robhillman97 committed:
added MRTK input system profiles

robhillman97 committed:
Added OpenXR plugin to link.xml in MRTK generated folder

robhillman97 committed:
Added new OpenXR interaction profile to fix interaction issues with right hand

robhillman97 committed:
Temporary work arounds for TutorialManager NRE and SpatialMapManager NRE. Also comments out debug log of moodle response in MoodleManager

robhillman97 committed:
Added world space console window for #1418 and fixed #1430

robhillman97 committed:
Fix for pick and place trigger settings closing #1432

robhillman97 committed:
fix for 1414

robhillman97 committed:
set interaction profile to legacy

robhillman97 committed:
set bit Depth submission to 24 bit

robhillman97 committed:
Fix for sketchfab model download

robhillman97 committed:
fix for tutorial on hololens

robhillman97 committed:
Updated Vuforia to 10.15.4

robhillman97 committed:
removed old vuforia files

robhillman97 committed:
resolving errors in image marker code after vuforia update

robhillman97 committed:
Added vuforia package file

robhillman97 committed:
Updates to ImageTargetManagerVuforia.cs

robhillman97 committed:
Updates to ImageTargetVuforia.cs

BorisJov committed:
Hololens tutorial fix

robhillman97 committed:
Fixes for image markers and calibration

robhillman97 committed:
Clean up image marker code

robhillman97 committed:
Update .gitattributes

robhillman97 committed:
Merge branch 'develop' of github.com:WEKIT-ECS/MIRAGE-XR into HololensFix

robhillman97 committed:
Fix for Brand Manager error after pull from develop

robhillman97 committed:
Vuforia image markers can be deleted

robhillman97 committed:
Vuforia Imagemarkers now working correctly

robhillman97 committed:
Cleaned up code after image marker changes

robhillman97 committed:
Merge branch 'HololensFix' into viewing_tutorial

robhillman97 committed:
Merge pull request #1443 from WEKIT-ECS/viewing_tutorial

Hololens fixes

robhillman97 committed:
Fixed issue with world space console preventing android and ios from building addressables

robhillman97 committed:
ARFoundation markers are now working again

robhillman97 committed:
Added vuforia licence key

robhillman97 committed:
Merge branch 'develop' of github.com:WEKIT-ECS/MIRAGE-XR into HololensFix

robhillman97 committed:
fixed error in brnad manager

cawke commented on Assets/MirageXR/Common/Scripts/CombinedEditor/ActionDetailView/Annotation Editors/ImageMarker Prefabs/ImageMarkerController.cs:
```diff
@@ -12,6 +13,7 @@ public class ImageMarkerController : MirageXRPrefab
 
         private string _imageName;
         private ToggleObject _content;
+        public IImageTarget target;
```
```suggestion
        public IImageTarget _target;
```

cawke commented on Assets/MirageXR/Common/Scripts/Managers/ImageTarget/Manager/ImageTargetManagerVuforia.cs:
```diff
@@ -9,19 +8,14 @@ public class ImageTargetManagerVuforia : ImageTargetManagerBase
+        while (!vuforiaRuntime.IsInitialized)
         {
             await Task.Yield();
         }
```
activate **delayed initialization** and call the initialization here.

![image](https://github.com/WEKIT-ECS/MIRAGE-XR/assets/20320355/406ba29f-c2c1-4198-a8e2-d22ee844d316)

```
        var vuforiaRuntime = VuforiaApplication.Instance;

        if (!vuforiaRuntime.IsInitialized)
        {
            vuforiaRuntime.Initialize();
            while (!vuforiaRuntime.IsInitialized)
            {
                await Task.Yield();
            }
        }
```

cawke commented on Assets/MirageXR/Common/Scripts/Managers/ImageTarget/Target/ImageTargetVuforia.cs:
```diff
@@ -9,27 +9,36 @@ public class ImageTargetVuforia : ImageTargetBase
 
     public ImageTargetBehaviour imageTargetBehaviour => _imageTargetBehaviour;
 
+    private Status _oldTargetStatus;
```
```
public class ImageTargetVuforia : ImageTargetBase
{
    private ImageTargetBehaviour _imageTargetBehaviour;
    private Status _oldTargetStatus;

    public ImageTargetBehaviour imageTargetBehaviour => _imageTargetBehaviour;

    protected override void TrackerInitialization()
    {
        _imageTargetBehaviour = GetComponent<ImageTargetBehaviour>();
        _oldTargetStatus = _imageTargetBehaviour.TargetStatus.Status;
        _imageTargetBehaviour.OnTargetStatusChanged += OnTrackableStatusChangedAction;
    }

    private void OnTrackableStatusChangedAction(ObserverBehaviour observerBehaviour, TargetStatus targetStatusChangeResult)
    {
        var statusChangeResult = targetStatusChangeResult.Status;
        OnStateChanged(ToTrackingState(_oldTargetStatus), ToTrackingState(statusChangeResult));
        _oldTargetStatus = statusChangeResult;
    }

    private static TrackingState ToTrackingState(Status state)
    {
        return state switch
        {
            Status.NO_POSE => TrackingState.Lost,
            Status.LIMITED => TrackingState.Limited,
            Status.EXTENDED_TRACKED => TrackingState.Limited,
            Status.TRACKED => TrackingState.Found,
            _ => throw new ArgumentOutOfRangeException(nameof(state), state, null)
        };
    }
}
```

cawke commented on Assets/MirageXR/Player/Scripts/Managers/SpatialMapManager.cs:
```diff
@@ -34,19 +34,26 @@ private void OnDestroy()
+            {
+                if (spatialMapMesh)
+                    spatialMapMesh.enabled = !editMode;
+            }
```
```suggestion
                if (spatialMapMesh)
                {
                     spatialMapMesh.enabled = !editMode;
                }
            }
```

cawke commented on Assets/MirageXR/Player/Scripts/Moodle/Login.cs:
```diff
@@ -341,6 +341,11 @@ public void showLRSSettings()
             ShowPanel(lrsPanel);
         }
 
+        public void showDeveloperOptions()
```
```suggestion
        public void ShowDeveloperOptions()
```

cawke commented:
Since you moved LoginMenu to ActivitySelectionMenu, remove CanvasUtility, as this will cause an error.

can i know the reason for moving the prefab? i'm just working on a similar window.

![image](https://github.com/WEKIT-ECS/MIRAGE-XR/assets/20320355/77c16e88-a33b-4f46-9d63-f66270cde284)

robhillman97 committed:
fix for augmentations not moving with image marker

robhillman97 commented:
> Since you moved LoginMenu to ActivitySelectionMenu, remove CanvasUtility, as this will cause an error.
> 
> can i know the reason for moving the prefab? i'm just working on a similar window.

We were making this object a child in code in login.cs line 50 with:

`           // make tha login panel a child of the activity list

            transform.position = loginPanelPosition.position;

            transform.rotation = loginPanelPosition.rotation;

            loginCanvas.transform.SetParent(FindObjectOfType<ActivitySelectionMenu>().transform);`

and I wanted to add a reference to the new console window with the login panel without using GameObject.find(); so I added this object to the scene and removed the code in login.cs

robhillman97 committed:
Removed pointCloudManager initialization from hololens builds due to image marker crashes

robhillman97 committed:
Requested changes for #1170

robhillman97 committed:
fixed broken references on login menu after method name changes

cawke commented on Assets/MirageXR/Common/Scripts/Managers/ImageTarget/Manager/ImageTargetManagerVuforia.cs:
```diff
@@ -9,19 +8,16 @@ public class ImageTargetManagerVuforia : ImageTargetManagerBase
+        while (!vuforiaRuntime.IsInitialized)
         {
             await Task.Yield();
         }
```
add check `if (!vuforiaRuntime.IsInitialized)` before `vuforiaRuntime.Initialize();`

```
        var vuforiaRuntime = VuforiaApplication.Instance;

        if (!vuforiaRuntime.IsInitialized)
        {
            vuforiaRuntime.Initialize();
            while (!vuforiaRuntime.IsInitialized)
            {
                await Task.Yield();
            }
        }
```

robhillman97 committed:
Added check for if vuforia initialized to vuforia target manager

robhillman97 committed:
Changes to Hololens spatial awerness system

robhillman97 committed:
Removed updates to activity selection due to merge conflicts

robhillman97 committed:
Merged develop into HololensFix and resolved conflicts

robhillman97 committed:
re-added activity selection updates after develop merge

robhillman97 committed:
Changed MRTK spatial awareness system settings

robhillman97 committed:
Merge branch 'develop' of github.com:WEKIT-ECS/MIRAGE-XR into HololensFix

robhillman97 committed:
Removed sphere object
